### PR TITLE
update(panel): Add show/hide and comment that addClass/removeClass/toggleClass should not show/hide.

### PR DESCRIPTION
**Pull request by KarenParker**
@ErinCoughlan @gmoothart @DerekLouie @jelbourn 

Please review.

Fixes issue #7733.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -254,34 +253,22 @@ angular
 
 /**
  * @ngdoc method
- * @name MdPanelRef#addClass
```
Comment (jelbourn) [inline src/components/panel/panel.js]: Why not leave `addClass` et al here, since they're separately useful from show/hide?

Comment (KarenParker) [inline src/components/panel/panel.js]: Done

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -254,34 +253,49 @@ angular
...
  * @ngdoc method
  * @name MdPanelRef#removeClass
  * @description
- * Removes a class from the panel. This can be used to hide/show the panel.
+ * Removes a class from the panel. DO NOT use this to hide/show the panel.
  *
- * @param {string} oldClass Class to be removed.
- * @returns {!angular.$q.Promise} A promise that is resolved when the panel has
- * removed the old class and animations are completed.
+ * @param {string} odlClass Class to be removed.
```
Comment (jelbourn) [inline src/components/panel/panel.js]: `odl`

File: src/components/panel/panel.js [hunk 3]
```diff
@@ -738,29 +761,28 @@ MdPanelRef.prototype.detachOnly = function() {
...
- * Add a class to the panel. This can control the panel's visibility.
+ * Shows the panel.
  *
- * @param {string} newClass Class to be added.
  * @returns {!angular.$q.Promise} A promise that is resolved when the panel has
- * the new class and animations finish.
+ * shown and animations finish.
  */
-MdPanelRef.prototype.addClass = function(newClass) {
+MdPanelRef.prototype.show = function() {
```
Comment (jelbourn) [inline src/components/panel/panel.js]: Internally, don't `show` and `hide` simply need to call `addClass` and do some extra stuff?
```js
MdPanelRef.prototype.show = function() {
  return this.removeClass('internal-hidden-class').then( /** show backdrop? */ );
}
```

File: src/components/panel/panel.js [hunk 4]
```diff
@@ -769,59 +791,85 @@ MdPanelRef.prototype.addClass = function(newClass) {
...
+ *
+ * @param {string} newClass Class to be added.
+ */
+MdPanelRef.prototype.addClass = function(newClass) {
+  if (!this._panelContainer) {
+    throw new Error('Panel does not exist yet. Call open() or attach().');
+  }
+
+  !this._panelContainer.hasClass(newClass) &&
+  this._panelContainer.addClass(newClass);
```
Comment (jelbourn) [inline src/components/panel/panel.js]: This addClass function should no-op if the class is already there.

File: src/components/panel/panel.js (hunk 3 above)
Comment (KarenParker) [inline src/components/panel/panel.js]: Yes, but it doesn't return a promise.

File: src/components/panel/panel.js (hunk 4 above)
Comment (KarenParker) [inline src/components/panel/panel.js]: It does already.... I reformatted it for clarity.

**Comment by jelbourn** [comment]
LGTM

**Comment by gmoothart** [comment]
:+1:

**Comment by ErinCoughlan** [comment]
LGTM

**Pull request closed without merging**